npm/pkg/dataplane/ipsets/ipset.go

```diff
@@ -150,6 +151,53 @@ func (set *IPSet) GetSetContents() ([]string, error) {
 	}
 }
 
+// ShallowCompare check if the properties of IPSets are same
+func (set *IPSet) ShallowCompare(newSet *IPSet) bool {
```

Comment:
Where do we plan to use this? I removed similar code since we don't take in ipsets as parameters now (have to do calls like addreference, addmember, etc.). 

For reference, I wrote this beforehand. Your checks for ippodkey and memberipsets are a little off, and Evan and I prefer not having if-bool-return-bool code. 

```
func ShallowCompare(set1 *IPSet, set2 *IPSet) bool {
	return set1.Name == set2.Name &&
		set1.Type == set2.Type &&
		set1.Kind == set2.Kind
}

func AreSameSet(set1 *IPSet, set2 *IPSet) bool {
	return ShallowCompare(set1, set2) && 
		haveSameSetContents(set1, set2)
}

func haveSameSetContents(set1 *IPSet, set2 *IPSet) bool {
	return haveSameCounts(set1, set2) ||
		haveSameIPPodKeys(set1, set2) ||
		haveSameMemberIPsets(set1, set2)
}

func haveSameCounts(set1 *IPSet, set2 *IPSet) bool {
	return set1.ipsetReferCount == set2.ipsetReferCount && // remove refer count or add kernel refer count too?
		len(set1.NetPolReference) == len(set2.NetPolReference) &&
		len(set1.SelectorReference) == len(set2.SelectorReference) &&
		len(set1.IPPodKey) == len(set2.IPPodKey) &&
		len(set1.MemberIPSets) == len(set2.MemberIPSets)
}

// assumes sets have same number of IPs
func haveSameIPPodKeys(set1 *IPSet, set2 *IPSet) bool {
	for ip, podKey1 := range set1.IPPodKey {
		podKey2, exists := set2.IPPodKey[ip]
		if !exists || podKey1 != podKey2 {
			return false
		}
	}
	return true
}

// assumes sets have same number of members
func haveSameMemberIPsets(set1 *IPSet, set2 *IPSet) bool {
	for memberName := range set1.MemberIPSets {
		_, exists := set2.MemberIPSets[memberName]
		if !exists {
			return false
		}
	}
	return true
}
```

Reply:
Yes i was thinking in the same lines, we could calculate the HASH of a given struct to determine if they are same. But this overly complicates existing scenario of UPDATE policy. So i tweaked the design to now Remove existing cached policy and Apply the new updated policy. This keeps things super simple for us while determining what dependencies to take care of.